### PR TITLE
Feature/tpp 00084 Multi-select

### LINKED ISSUE #84: TPP-00084: Log Multiple Period Days Screen
> As a user, I want to be able to quickly log multiple days where I have had my period 
Screen Shot 2022-02-15 at 3 45 29 PM
Screen Shot 2022-02-15 at 4 42 17 PM

## **Technical Information**
- Tapping X on top left corner leads back to Calendar with changes UNSAVED
- Changes only Saved by pressing round checkmark button 
  - After clicking the Checkmark button to save, navigate back to Calendar and set the calendar's filter to FLOW so user can see reflected changes
  - Traverse through selected dates and set the flow attribute to MEDIUM. Even if a flow level was already logged, set it to MEDIUM
- Tapping an unselected date will select it with green colour and checkmark, Tapping a selected date will de-select it
- No limit to selections, user can select as many as they want
- Non-infinite scroll (scroll ends at current month), but future dates should be DISABLED using a light-grey colour. Recall we do not want user logging stuff for days in the future

## **Styling Requirements**
- White Gradient on the bottom
- Underneath "Tap date to log period", initially says "0 days selected", but with each selection updates to "# days selected" 
- We would like the calendar to not shift after clicking the checkmark to save. For example, if Feb 2022 was in view, clicking save means Feb 2022 is still the month in view and has not shifted (as in the second image, see how after finishing in the log multiple period dates and navigating back to the calendar, the calendar hasn't shifted?)

### PR DESCRIPTION
**Pull request by roylin2000**
# Description

Added a new screen that allows the user to select multiple period days at once. The main changes are to the calendar and the inclusion of the new multi-select screen. (I'll need to pull the PR with the view filter to reflect the changes after multi-select)

image

Fixes #84 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [ ] My changes generate no new warnings (Could not locate shadow view with tag #4449, this is probably caused by a temporary inconsistency between native views and shadow views.) Not sure how to resolve this yet.
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (kenneth-miura) [inline tpp-app/src/home/pages/CalendarScreen.js]: isn't this equivalent to just writing `style={styles.dayContainer}`?

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (kenneth-miura) [inline tpp-app/src/home/pages/CalendarScreen.js]: When does `navigation` change such that it triggers the useEffect? I'm not super familiar with how the navigation library works

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (kenneth-miura) [inline tpp-app/src/home/pages/CalendarScreen.js]: I think u can get current year from a listener on the calendar? Edward did it for https://github.com/uoftblueprint/the-period-purse/pull/108

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (kenneth-miura) [inline tpp-app/src/home/pages/CalendarScreen.js]: They're both objects so JS will check if they are the same object in memory, so I think `data != calendarData` is just always true

File: tpp-app/src/services/LogSymptomsService.js
Comment (kenneth-miura) [inline tpp-app/src/services/LogSymptomsService.js]: doesn't really make sense to use both `await` and `.then` syntax. Imo just use '.then'

File: tpp-app/src/services/LogSymptomsService.js
Comment (kenneth-miura) [inline tpp-app/src/services/LogSymptomsService.js]: I think it would be nicer to write a helper function with argument `year` & call it three times here, since there's repeated code

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
Looks great! Got some code style stuff but aside from that everything is well written & makes sense

**Review by helen-li — APPROVED** [review]
looks good to me!! the one suggestion that i had kenneth already mentioned (adding the helper) so i will just approve here ☺️

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (faye-tan) [reply tpp-app/src/home/pages/CalendarScreen.js]: @edwardhuahan I think you should take a look at this part to make sure it works with what you did with the filters

File: tpp-app/src/home/pages/LogMultipleDatesScreen.js
Comment (faye-tan) [inline tpp-app/src/home/pages/LogMultipleDatesScreen.js]: suggest remove big commented out blocks  if not using

File: tpp-app/src/home/pages/LogMultipleDatesScreen.js
Comment (faye-tan) [inline tpp-app/src/home/pages/LogMultipleDatesScreen.js]: suggest remove big commented out blocks if not using

File: tpp-app/src/home/pages/LogMultipleDatesScreen.js
Comment (faye-tan) [inline tpp-app/src/home/pages/LogMultipleDatesScreen.js:190]: 👏

File: tpp-app/src/services/LogSymptomsService.js
Comment (roylin2000) [reply tpp-app/src/services/LogSymptomsService.js]: I'll do this after we merge. No time rn :(

File: tpp-app/src/services/LogSymptomsService.js
Comment (roylin2000) [reply tpp-app/src/services/LogSymptomsService.js]: True!

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/CalendarScreen.js]: Oh right, I'll just remove it.

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/CalendarScreen.js]: oh yea, fixed it!

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/CalendarScreen.js]: It should trigger when it is used. At least that's what I thought.

File: tpp-app/src/home/pages/LogMultipleDatesScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/LogMultipleDatesScreen.js]: true true, removed!

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/CalendarScreen.js]: I will ask edward about that, but for now we can just merge since we have no time lol.

Comment (kenneth-miura) [reply tpp-app/src/home/pages/CalendarScreen.js]: sure. Just make sure @faye-tan knows so we add it as an issue on the todo issue

File: tpp-app/src/services/LogSymptomsService.js
Comment (kenneth-miura) [reply tpp-app/src/services/LogSymptomsService.js]: sure. Just make sure @faye-tan knows so we add it as an issue on the todo issue

File: tpp-app/src/services/utils/helpers.js
Comment (kenneth-miura) [inline tpp-app/src/services/utils/helpers.js:108]: ah you are right this doesn't need to be async. I'd say don't make the swap in this PR, and open an issue for removing the async keyword & updating all uses. Since this function gets called in a lot of different places, and also in other PRs that are currently open (mine, Emily's at least). 

What do you think @faye-tan

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
I don't think you pushed the commit fixing some of the PR comments. The other changes look great tho.

I added one comment about removing async on `getSymptomsFromCalendar`. That's my b lol I think I was on autopilot for that function. I tagged Faye on it since it's a ghub issues type question. lmk what you think as well

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (faye-tan) [reply tpp-app/src/home/pages/CalendarScreen.js]: I'm ok with leaving it for later, but @roylin2000  I'll assign you to go back in and make changes then.

File: tpp-app/src/services/LogSymptomsService.js
Comment (faye-tan) [reply tpp-app/src/services/LogSymptomsService.js]: Same comment a above, ok with leaving it for later, but @roylin2000 I'll assign you to go back in and make changes later then.

File: tpp-app/src/home/pages/CalendarScreen.js
Comment (roylin2000) [reply tpp-app/src/home/pages/CalendarScreen.js]: yup that sounds good!

File: tpp-app/src/services/LogSymptomsService.js
Comment (roylin2000) [reply tpp-app/src/services/LogSymptomsService.js]: gotcha!

**Comment by roylin2000** [comment]
> I don't think you pushed the commit fixing some of the PR comments. The other changes look great tho.
> 
> I added one comment about removing async on `getSymptomsFromCalendar`. That's my b lol I think I was on autopilot for that function. I tagged Faye on it since it's a ghub issues type question. lmk what you think as well

Oh my god, my bad LOL. I have now pushed the commits.

File: tpp-app/src/services/utils/helpers.js
Comment (faye-tan) [reply tpp-app/src/services/utils/helpers.js:108]: I'm ok with changing it here, and I'll go through all the other PRs to ask you and Emily to change it before you merge since its true it doesn't have to be async

Comment (roylin2000) [reply tpp-app/src/services/utils/helpers.js:108]: ahh I like that suggestion. It was just that my code was bugging when it was async for some reason. I'll try it out with it back as an async function.

Comment (roylin2000) [reply tpp-app/src/services/utils/helpers.js:108]: oh ok!

**Review by kenneth-miura — APPROVED** [review]
lgtm!

**Pull request merged**
